lib/iris/cube.py

```diff
@@ -1694,7 +1695,8 @@ def data(self):
                 msg = msg.format(self.shape, data.dtype)
                 raise MemoryError(msg)
             # Unmask the array only if it is filled.
-            if isinstance(data, np.ndarray) and ma.count_masked(data) == 0:
+            if (isinstance(data, np.ma.masked_array) and
+                    ma.count_masked(data) == 0):
```

Comment:
At the moment I'm just totally ignoring the masked issue.
This code is effectively a dead branch with the latest changes, as ```as_concrete_data``` will never return a masked result.
But this will need fixing later, so I left it.